packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -246,12 +254,117 @@ describe("SharedTree", () => {
         // Check that the edit is reflected in the EditableTree after the transaction.
         assert.equal(editable[valueSymbol], 2);
     });
+
+    it("can insert optional child field", async () => {
+        const childKey: FieldKey = brand("optionalChild");
+        const value = "42";
+        const provider = await TestTreeProvider.create(2);
+        const [tree1, tree2] = provider.trees;
+
+        initializeTestTreeWithValue(tree1, 1);
+        await provider.ensureSynchronized();
+
+        assert(isUnwrappedNode(tree1.root));
+        const anchor = tree1.root[anchorSymbol];
+        tree1.runTransaction((forest, editor) => {
+            tree1.context.prepareForEdit();
+            const field = editor.optionalField(tree1.locate(anchor), childKey);
+            const writeCursor = singleTextCursor({ type: brand("TestValue"), value });
+            field.set(writeCursor, true);
+            return TransactionResult.Apply;
+        });
+
+        assert(childKey in tree1.root);
+        const child = tree1.root[childKey];
+        assert(isUnwrappedNode(child));
+        assert.equal(child[valueSymbol], value);
+        assert.equal(child[getTypeSymbol](), "TestValue");
+
+        await provider.ensureSynchronized();
+        assert(isUnwrappedNode(tree2.root));
+        assert(childKey in tree2.root);
+        const child2 = tree2.root[childKey];
+        assert(isUnwrappedNode(child2));
+        assert.equal(child2[valueSymbol], value);
+        tree1.context.free();
+        tree2.context.free();
+    });
+
+    it("can insert value child field", async () => {
+        const childKey: FieldKey = brand("valueChild");
+        const value = "42";
+        const provider = await TestTreeProvider.create(2);
+        const [tree1, tree2] = provider.trees;
+
+        const hasNoFields = namedTreeSchema({
+            name: brand("HasNoFields"),
+            extraLocalFields: fieldSchema(FieldKinds.sequence),
+        });
+        const hasValueField = namedTreeSchema({
+            name: brand("HasValueField"),
+            localFields: {
+                valueChild: fieldSchema(FieldKinds.value, [hasNoFields.name]),
```

Comment:
if we are gong to have a string constant for this, we should use it consistently.
```suggestion
                [childKey]: fieldSchema(FieldKinds.value, [hasNoFields.name]),
```